### PR TITLE
Array.Parallel.forall  , Array.Parallel.exists

### PR DESCRIPTION
**Pull request by T-Gro**
This is another part of the [RFC](https://github.com/fsharp/fslang-design/blob/ff115e7f10993101c1624bfdc5dad9376d64553c/RFCs/1130-array-parallel-collection-functions-more-regular.md) to expand the Array.Parallel module, based on https://github.com/fsharp/fslang-suggestions/issues/187 .

There are two functions for logical tests, forall  and exists.
- They can fully step if a matching (resp. not matching for exists) element is found
- They do not need to return the first index (unlike tryFind etc.), therefore ParallelLoopState.Stop() can be used instead of .Break()
- No synchronization across worker threads is needed in this case

### CODE REVIEW

File: src/FSharp.Core/array.fsi [hunk 1]
```diff
@@ -3094,6 +3094,67 @@ module Array =
...
+        /// [1; 2] |> Array.Parallel.forall isEven // evaluates to false
+        /// </code>
+        /// </example>
+        [<CompiledName("ForAll")>]
+        [<Experimental("Experimental library feature, requires '--langversion:preview'")>]
+        val forall: predicate:('T -> bool) -> array:'T[] -> bool
+
+        /// <summary>Tests if any element of the array satisfies the given predicate.</summary>
+        ///
+        /// <remarks>The predicate is applied to the elements of the input array in paralll. If any application
```
Comment (psfinaki) [inline src/FSharp.Core/array.fsi:3126]:
```suggestion
        /// <remarks>The predicate is applied to the elements of the input array in parallel. If any application 
```

File: src/FSharp.Core/array.fsi [hunk 2]
```diff
@@ -3094,6 +3094,67 @@ module Array =
...
+        /// </example>
+        [<CompiledName("ForAll")>]
+        [<Experimental("Experimental library feature, requires '--langversion:preview'")>]
+        val forall: predicate:('T -> bool) -> array:'T[] -> bool
+
+        /// <summary>Tests if any element of the array satisfies the given predicate.</summary>
+        ///
+        /// <remarks>The predicate is applied to the elements of the input array in paralll. If any application 
+        /// returns true then the overall result is true and testing of other elements in all threads is stopped at system's earliest convenience.
+        /// Otherwise, false is returned.</remarks>
```
Comment (psfinaki) [inline src/FSharp.Core/array.fsi:3128]:
```suggestion
        /// Otherwise, <c>false</c> is returned.</remarks>
```

Comment (psfinaki) [reply src/FSharp.Core/array.fsi:3128]: and elsewhere, I guess

**Review by psfinaki — APPROVED** [review]

**Review by abonie — APPROVED** [review]

**Review by KevinRansom — APPROVED** [review]

**Pull request merged**